lexik/jwt-authentication-bundle/2.3/manifest.json

```diff
@@ -3,13 +3,14 @@
         "Lexik\\Bundle\\JWTAuthenticationBundle\\LexikJWTAuthenticationBundle": ["all"]
     },
     "copy-from-recipe": {
-        "config/": "%CONFIG_DIR%"
+        "config/": "%CONFIG_DIR%",
+        "secrets/": "secrets"
     },
     "aliases": ["jwt-auth"],
     "env": {
         "#1": "Key paths should be relative to the project directory",
-        "JWT_PRIVATE_KEY_PATH": "%CONFIG_DIR%/jwt/private.pem",
-        "JWT_PUBLIC_KEY_PATH": "%CONFIG_DIR%/jwt/public.pem",
+        "JWT_PRIVATE_KEY_PATH": "secrets/jwt/private.pem",
+        "JWT_PUBLIC_KEY_PATH": "secrets/jwt/public.pem",
```

Comment:
Shoudn't it be `%PROJECT_DIR%/secrets/jwt/...`

Reply:
No, unless we want to support storing the key pair outside of the project directory?

Reply:
But if we go ahead with this, we should probably add a `secrets-dir` in Flex?

Reply:
I agree with Fabien. This looks really strange. It should be `%CONFIG_DIR%/secrets/jwt/...` or `%PROJECT_DIR%/secrets/jwt/...`

Reply:
`%CONFIG_DIR%` resolves to `config`.
`%PROJECT_DIR%` resolves to the full path of the project directory.

What we might want to add is a `%SECRETS_DIR%` which resolves to `secrets`. But I still find it weird to see a `%SECRETS_DIR%` in an environment variable...

Reply:
In Rails 5 they introduced full management of secrets. They are stored encrypted in a file called `secrets.yml`. The contents of the file are separated by environment:

```yaml
# secrets.yml

shared: # <-- this one means, "all environments"
    # ...

development:
    # ...

test:
    # ...

production:
    # ...
```

Maybe we can introduce something like that for Symfony 4.x.

Reply:
symfony/symfony#23901 introduced the `file` prefix.

Reply:
@javiereguiluz managing secret's encryption at this level looks like a bad idea to me. There are already plenty dedicated tools to do it like Docker and Kubernetes. IMO the best thing to to is to make it easy to use them in Symfony (and it's achieved with symfony/symfony#23901 ), not to reinvent the wheel.

I'm 👍 to introduce a new `secrets` directory and store things like JWT keys and Doctrine passwords in it (and retrieve them with symfony/symfony#23901).

Reply:
The risk of the `secrets` directory is that people would be tempted to directly commit their credentials in the code repository... You use the example of Docker/Kubernetes and the secrets management is exactly meant to prevent people to add their secrets within the Docker image directly. As long as we give a very clear documentation that these secrets have to be development secrets and overwritten by production secrets while deploying production, then I'm 👍 with such `secrets` directory.

Reply:
Adding an empty directory with a `.gitignore` which ignores everything else should help.

Reply:
We can add this directory to the .gitignore file.